core/src/main/scala/org/ensime/indexer/lucene/SimpleLucene.scala

```diff
@@ -43,25 +54,22 @@ object SimpleLucene {
  * filtering rules based on tags.
  */
 class SimpleLucene(path: Path, analyzers: Map[String, Analyzer]) extends SLF4JLogging {
+  import SimpleLucene._
+
   Files.createDirectories(path)
 
+  private val executorService = Executors.newFixedThreadPool(8)
```

Comment:
right ok, so you put the EC here and then the IndexService is just for combining results, I see. Ok.